pytwis/pytwis_clt.py

```diff
@@ -593,28 +532,99 @@ def pytwis_cli():
 
     try:
         if len(args.socket) > 0:
-            twis = Pytwis(socket=args.socket, db=args.db, password=args.password)
+            pytwis = Pytwis(socket=args.socket, db=args.db, password=args.password)
         else:
-            twis = Pytwis(hostname=args.hostname, port=args.port, db=args.db, password=args.password)
+            pytwis = Pytwis(hostname=args.hostname, port=args.port, db=args.db, password=args.password)
+        return pytwis, prompt
+
     except ValueError as e:
         print('Failed to connect to the Redis server: {}'.format(str(e)),
               file=sys.stderr)
-        return -1
+        return None, None
+
+def pytwis_clt():
+    """The main routine of this command-line tool."""
+    epilog = '''After launching `pytwis_clt.py`, you will be able to use the following commands:
+
+    * Register a new user:
+
+        127.0.0.1:6379> register {username} {password}
+
+    * Log into a user:  
+
+        127.0.0.1:6379> login {username} {password} 
+
+    * Log out of a user:
+
+        127.0.0.1:6379> logout
+
+    * Change the password:
+
+        127.0.0.1:6379> changepwd {old_password} {new_password} {confirmed_new_password}
+
+    * Get the profile of the current user:
+
+        127.0.0.1:6379> userprofile
+
+    * Post a tweet:
+
+        127.0.0.1:6379> post {tweet}
+
+    * Follow a user:
+
+        127.0.0.1:6379> follow {followee_username}
+
+    * Unfollow a user:
+
+        127.0.0.1:6379> unfollow {followee_username}
+
+    * Get the follower list:
+
+        127.0.0.1:6379> followers
+
+    * Get the following list:
+
+        127.0.0.1:6379> followings
+
+    * Get the timeline:
+
+        127.0.0.1:6379> timeline
+        127.0.0.1:6379> timeline {max_tweet_count}
+
+    Note that if a user is logged in, `timeline` will return the user timeline; 
+    otherwise `timeline` will return the general timeline.
+
+    * Get the tweets posted by a user:
+
+        127.0.0.1:6379> tweetsby 
+        127.0.0.1:6379> tweetsby {username}
+        127.0.0.1:6379> tweetsby {username} {max_tweet_count}
+
+    Note that if no username is given, `tweetsby` will return the tweets posted 
+    by the currently logged-in user.
+
+    * Exit the program:
+
+        127.0.0.1:6379> exit
+        127.0.0.1:6379> quit
+    '''
+    pytwis, prompt = get_pytwis(epilog)
+    if pytwis == None:
+        return
```

Comment:
Return an error value like -1?

Reply:
Do you want to have three returns? error, pytwis, prompt?

Reply:
No, I mean that we return -1 if pytwis == None.

Reply:
I don't think that's good idea. If the first return value is the instance of Pytwis, then it must be None for error case, not an integer value, -1.

Reply:
Clarified my point offline.